R/telemetry.R

```diff
@@ -508,20 +514,54 @@ Telemetry <- R6::R6Class( # nolint object_name_linter
         checkmate::check_r6(session, "ShinySession", null.ok = TRUE),
         checkmate::check_class(session, "session_proxy")
       )
-
       self$data_storage$insert(
         app_name = self$app_name,
         type = type,
         session = session$token,
         details = details
       )
     },
-
+    .config_input_tracker = function(include_input_ids = NULL,
+                                     excluded_inputs = NULL,
+                                     excluded_inputs_regex = NULL,
+                                     track_values = FALSE) {
+      checkmate::assert_character(include_input_ids, null.ok = TRUE, any.missing = FALSE)
+      checkmate::assert_character(excluded_inputs, null.ok = TRUE, any.missing = FALSE)
+      checkmate::assert_character(excluded_inputs_regex, null.ok = TRUE, any.missing = FALSE)
+      checkmate::assert_logical(track_values, len = 1)
+      excluded_pattern <- ""
+      included_pattern <- ""
+      if (!is.null(excluded_inputs_regex)) {
+        excluded_pattern_regex <- paste(excluded_inputs_regex, collapse = "|")
+      } else {
+        excluded_pattern_regex <- ""
+      }
+      if (!is.null(excluded_inputs)) {
+        escaped_excludes <- sapply(excluded_inputs, function(input) {
+          gsub("([][{}()+*^$|\\\\?.^-])", "\\\\\\1", input)
+        })
+        excluded_pattern_exact <- paste(escaped_excludes, collapse = "|")
+      } else {
+        excluded_pattern_exact <- ""
+      }
+      excluded_pattern <- paste0(excluded_pattern_regex, "|", excluded_pattern_exact)
+      if (!is.null(include_input_ids)) {
+        included_pattern <- paste0("^(", paste(sapply(include_input_ids,
+                                                      gsub,
+                                                      pattern = "([][{}()+*^$|\\\\?.^-])",
+                                                      replacement = "\\\\\\1", fixed = FALSE),
+                                               collapse = "|"), ")$")
+        purrr::walk(include_input_ids, ~ self$log_input(input_id = .x, track_value = track_values))
```

Comment:
I'm not a big fan of doing this here.

Consider adding a new return element on the list with manual inputs ids and do it on the caller body.